### ISSUE TITLE
Running in Python 2.7.15 results in error in str

### ISSUE THREAD

**Issue by jonespm** [IMSGlobal/caliper-python#138]
There's an error I'm getting from python2

```
Traceback (most recent call last):
  File "caliper_tester.py", line 39, in <module>
    api_key=token)
  File "/usr/local/lib/python2.7/site-packages/caliper/base.py", line 248, in __init__
    self.API_KEY = api_key
  File "/usr/local/lib/python2.7/site-packages/caliper/base.py", line 167, in API_KEY
    raise_with_traceback(ValueError('new key value must be a string'))
  File "<string>", line 7, in raise_with_traceback
ValueError: new key value must be a string
```

I changed it to string_types and seems to be working . . .
https://stackoverflow.com/a/11301392/3708872

**Comment by ViktorHaag** [comment]
Given that Python 2 is going out of support after this year, I wonder whether it might just be better to abandon Python2 for this project? Thoughts on that?

(The Python2 in my test environment is Python 2.7.10 on Darwin, which perhaps explains why this wasn't caught in my testing...)

**Comment by ViktorHaag** [comment]
Also; I don't use `six`, I use `future` for compatibility instead, so a solution needs to work with `future`. I will investigate this...

**Comment by ViktorHaag** [comment]
@jonespm -- I think I see what's going on. The Caliper package code itself uses future to enforce "unicode_literals", meaning this code will make `a` a unicode string on both Py2 and Py3:
```
from __future__ import unicode_literals
a = '1234'
```

This works in the package's test runner code, because it, too, uses this convention and so it's scaffolding of an options class builds it with a unicode string.

Notice what happens in the Py2 repl:
```
>>> import caliper
>>> o = caliper.HttpOptions(api_key='1234')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/data/Dropbox/Work/IMS/Caliper/caliper-python/caliper/base.py", line 307, in __init__
    self.API_KEY = api_key
  File "/usr/local/data/Dropbox/Work/IMS/Caliper/caliper-python/caliper/base.py", line 213, in API_KEY
    raise_with_traceback(ValueError('new key value must be a string'))
  File "<string>", line 7, in raise_with_traceback
ValueError: new key value must be a string
>>> o = caliper.HttpOptions(api_key=u'1234')
>>> 
```

I suspect the simplest and best solution here is to document that the Caliper package expects that, whenever you pass a string to it, it's a unicode string. We can change all the exceptions that do testing on `str` to have messages that make this plain.

Thoughts on that proposal?

**Comment by jonespm** [comment]
I think we need to support Python 2 until it's retired, which is according to the clock not until the end of 2019. https://pythonclock.org/

Unfortunately we have 1-2 projects that I'm testing this with that we plan to migrate to Python 3 this summer, but if it can work in 2 with a simple change for now I think it's worthwhile.

I submitted a PR #139  that doesn't use six, it uses future.utils. I'm fine if you'd rather just document it instead though. 

The code I used to test this with is at 
https://github.com/jonespm/python-caliper-tester

It's basically the snippet in the README that runs against the lti.tools test endpoint so it's self contained but I was also testing OpenLRW with it.
